Errors in console when manually editing slider values

Errors appear in console when you use the text box to edit a slider value.

```
[2023-09-14 19:10:37] [VisionModule - VisionModuleChangeSubscriber - Integrated_Camera] [ERROR] Failed to invoke camera settable method: setExposure: null
[2023-09-14 19:10:37] [VisionModule - VisionModuleChangeSubscriber - Integrated_Camera] [ERROR] java.lang.IllegalArgumentException
        at jdk.internal.reflect.GeneratedMethodAccessor4.invoke(Unknown Source)
        at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
        at java.base/java.lang.reflect.Method.invoke(Method.java:568)
        at org.photonvision.vision.processes.VisionModuleChangeSubscriber.onDataChangeEvent(VisionModuleChangeSubscriber.java:180)
        at org.photonvision.common.dataflow.DataChangeService.dispatchFromQueue(DataChangeService.java:65)
        at java.base/java.lang.Thread.run(Thread.java:833)

[2023-09-14 19:10:37] [VisionModule - VisionModuleChangeSubscriber - Integrated_Camera] [ERROR] Unknown exception when setting PSC prop!: class java.lang.String cannot be cast to class java.lang.Number (java.lang.String and java.lang.Number are in module java.base of loader 'bootstrap')
[2023-09-14 19:10:37] [VisionModule - VisionModuleChangeSubscriber - Integrated_Camera] [ERROR] java.lang.ClassCastException: class java.lang.String cannot be cast to class java.lang.Number (java.lang.String and java.lang.Number are in module java.base of loader 'bootstrap')
        at org.photonvision.vision.processes.VisionModuleChangeSubscriber.onDataChangeEvent(VisionModuleChangeSubscriber.java:204)
        at org.photonvision.common.dataflow.DataChangeService.dispatchFromQueue(DataChangeService.java:65)
        at java.base/java.lang.Thread.run(Thread.java:833)
```

--- a/photon-client/src/components/cameras/CameraSettingsCard.vue
+++ b/photon-client/src/components/cameras/CameraSettingsCard.vue
@@ -36,7 +36,7 @@ const saveCameraSettings = () => {
         });
       }
     });
-  };
+};
 
 watchEffect(() => {
   currentFov.value = useCameraSettingsStore().currentCameraSettings.fov.value;

--- a/photon-client/src/components/cameras/CameraSettingsCard.vue
+++ b/photon-client/src/components/cameras/CameraSettingsCard.vue
@@ -3,20 +3,22 @@ import CvSelect from "@/components/common/cv-select.vue";
 import CvNumberInput from "@/components/common/cv-number-input.vue";
 import { useCameraSettingsStore } from "@/stores/settings/CameraSettingsStore";
 import { useStateStore } from "@/stores/StateStore";
-import { ref } from "vue";
+import { ref, watchEffect } from "vue";
 
-const currentFov = ref(useCameraSettingsStore().currentCameraSettings.fov.value);
+const currentFov = ref();
 
 const saveCameraSettings = () => {
   useCameraSettingsStore()
-    .updateCameraSettings({ fov: currentFov.value }, true)
+    .updateCameraSettings({ fov: currentFov.value }, false)
     .then((response) => {
+      useCameraSettingsStore().currentCameraSettings.fov.value = currentFov.value;
       useStateStore().showSnackbarMessage({
         color: "success",
         message: response.data.text || response.data
       });
     })
     .catch((error) => {
+      currentFov.value = useCameraSettingsStore().currentCameraSettings.fov.value;
       if (error.response) {
         useStateStore().showSnackbarMessage({
           color: "error",
@@ -33,11 +35,12 @@ const saveCameraSettings = () => {
           message: "An error occurred while trying to process the request."
         });
       }
-    })
-    .finally(() => {
-      useCameraSettingsStore().currentCameraSettings.fov.value = currentFov.value;
     });
-};
+  };
+
+watchEffect(() => {
+  currentFov.value = useCameraSettingsStore().currentCameraSettings.fov.value;
+});
 </script>
 
 <template>

--- a/photon-client/src/components/common/cv-range-slider.vue
+++ b/photon-client/src/components/common/cv-range-slider.vue
@@ -33,7 +33,7 @@ const localValue = computed<[number, number]>({
     return Object.values(props.value) as [number, number];
   },
   set: (v) => {
-    for(let i = 0; i < v.length; i++) {
+    for (let i = 0; i < v.length; i++) {
       v[i] = parseFloat(v[i] as unknown as string);
     }
     emit("input", v);

--- a/photon-client/src/components/common/cv-range-slider.vue
+++ b/photon-client/src/components/common/cv-range-slider.vue
@@ -32,7 +32,12 @@ const localValue = computed<[number, number]>({
   get: (): [number, number] => {
     return Object.values(props.value) as [number, number];
   },
-  set: (v) => emit("input", v)
+  set: (v) => {
+    for(let i = 0; i < v.length; i++) {
+      v[i] = parseFloat(v[i] as unknown as string);
+    }
+    emit("input", v);
+  }
 });
 
 const changeFromSlot = (v: number, i: number) => {

--- a/photon-client/src/components/common/cv-slider.vue
+++ b/photon-client/src/components/common/cv-slider.vue
@@ -27,7 +27,7 @@ const emit = defineEmits<{
 
 const localValue = computed({
   get: () => props.value,
-  set: (v) => emit("input", v)
+  set: (v) => emit("input", parseFloat(v as unknown as string))
 });
 </script>
 

--- a/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
+++ b/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
@@ -333,7 +333,7 @@ public class RequestHandler {
             ctx.status(200);
             ctx.result("Successfully saved camera settings");
             logger.info("Successfully saved camera settings");
-        } catch (JsonProcessingException|NullPointerException e) {
+        } catch (JsonProcessingException | NullPointerException e) {
             ctx.status(400);
             ctx.result("The provided camera settings were malformed");
             logger.error("The provided camera settings were malformed", e);

--- a/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
+++ b/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
@@ -333,7 +333,7 @@ public class RequestHandler {
             ctx.status(200);
             ctx.result("Successfully saved camera settings");
             logger.info("Successfully saved camera settings");
-        } catch (JsonProcessingException e) {
+        } catch (JsonProcessingException|NullPointerException e) {
             ctx.status(400);
             ctx.result("The provided camera settings were malformed");
             logger.error("The provided camera settings were malformed", e);

--- a/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
+++ b/photon-server/src/main/java/org/photonvision/server/RequestHandler.java
@@ -323,7 +323,7 @@ public class RequestHandler {
             var data = kObjectMapper.readTree(ctx.body());
 
             int index = data.get("index").asInt();
-            double fov = kObjectMapper.readTree(data.get("settings").asText()).get("fov").asDouble();
+            double fov = data.get("settings").get("fov").asDouble();
 
             var module = VisionModuleManager.getInstance().getModule(index);
             module.setFov(fov);

